lib/mongo/server/connection.rb

```diff
@@ -190,9 +190,12 @@ def handshake!
 
       def authenticate!
         if options[:user] || options[:auth_mech]
-          user = Auth::User.new(Options::Redacted.new(:auth_mech => default_mechanism).merge(options))
+          user = Auth::User.new(Options::Redacted.new(:auth_mech => default_mechanism).merge(options), @client_key)
```

Comment:
since everything is already passed to the user initialize method as part of the options hash, if we want to be consistent, we could make this:

```ruby
user = Auth::User.new(Options::Redacted.new(:auth_mech => default_mechanism, :client_key => @client_key).merge(options))
```

and then extract the client key in the #initialize method of the user:
```ruby
@client_key = options[:client_key]
```